lib/oli/delivery.ex

```diff
@@ -291,4 +292,43 @@ defmodule Oli.Delivery do
         {:ok, section}
     end
   end
+
+  def has_completed_survey?(section_slug, user_id) do
+    survey = Sections.get_survey(section_slug)
```

Comment:
This approach to checking if all of the activities present are evaluated is problematic because it does not take into account activity bank selections.   Furthermore, it only identifies `activity-reference` elements at the top level (they can exist nested within groups)

Instead of fetching the page revision, the correct approach is to simply fetch the `%ActivityAttempt` records for that student for the latest `%ResourceAttempt` for the survey resource id.  It is crucial that only the latest `%ResourceAttempt` record is considered.  For the latest resource attempt, if all activity records present are evaluated, then we consider this required survey page to be complete.   This approach properly takes into account activity bank selections (and will handle the upcoming audience filtered content feature).